### PR TITLE
Add limit to merge function for KHyperLogLog

### PR DESCRIPTION
**Pull request by feilong-liu**
## Description

After https://github.com/prestodb/presto/pull/21510, the constructor for `KHyperLogLogStateFactory` takes a long integer as input for group limit. However, the merge function for KHyperLogLog still uses the parametric aggregation style, which makes the new constructor not valid, hence throw compile error. It can be solved with two ways: 1) simply add a default constructor without any input parameter. However, it will make the merge function facing the same memory issue we solved for khyperloglog_agg 2) make the merge function to not use parametric aggregation style

In this PR, I apply both approaches, i.e. add a default constructor without any input parameter for `KHyperLogLogStateFactory` so that the current function will not fail compilation. I also add another version of implementation for the `merge` function of khyperloglog which can take group limit. Which version to use is controlled by feature config, so that it can be rolled out smoothly.

Since I implement two versions for merge khyperloglog, I also chose to improve the change in https://github.com/prestodb/presto/pull/21510 so that it also has two versions, and gated by the same config.

## Motivation and Context
Similar to https://github.com/prestodb/presto/pull/21510
In addition to the memory issue, the merge function for KHyperLogLog currently does not compile, it will fix the problem too.

## Impact
Refer to https://github.com/prestodb/presto/pull/21510.

## Test Plan
Add e2e test, also run verifier locally

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

General Changes
* Fix the compilation error of merge function for KHyperLogLog state
* Add an feature config property `limit-khyperloglog-agg-group-number-enabled` to control whether to limit the number of groups for aggregation functions using KHyperLogLog state
```

### CODE REVIEW

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 1]
```diff
@@ -7419,4 +7419,18 @@ public void testReplaceConstantVariableReferencesWithConstants()
...
+                        "(2, 1, 11, 30), (2, 1, 11, 11), (2, 2, 9, 1), (2, 2, 87, 2)) t(k1, k2, v1, v2) group by k1, k2) group by k1",
+                "select k1, cardinality(khyperloglog_agg(v1, v2)), uniqueness_distribution(khyperloglog_agg(v1, v2)) from (values (1, 1, 2, 3), (1, 1, 4, 0), (1, 2, 90, 20), (1, 2, 87, 1), (2, 1, 11, 30), (2, 1, 11, 11), " +
+                        "(2, 2, 9, 1), (2, 2, 87, 2)) t(k1, k2, v1, v2) group by k1");
+
+        assertQueryWithSameQueryRunner("select cardinality(merge(khll)), uniqueness_distribution(merge(khll)) from (select k1, k2, khyperloglog_agg(v1, v2) khll from (values (1, 1, 2, 3), (1, 1, 4, 0), (1, 2, 90, 20), (1, 2, 87, 1), " +
+                        "(2, 1, 11, 30), (2, 1, 11, 11), (2, 2, 9, 1), (2, 2, 87, 2)) t(k1, k2, v1, v2) group by k1, k2)",
+                "select cardinality(khyperloglog_agg(v1, v2)), uniqueness_distribution(khyperloglog_agg(v1, v2)) from (values (1, 1, 2, 3), (1, 1, 4, 0), (1, 2, 90, 20), (1, 2, 87, 1), (2, 1, 11, 30), (2, 1, 11, 11), " +
+                        "(2, 2, 9, 1), (2, 2, 87, 2)) t(k1, k2, v1, v2)");
+    }
```
Comment (ajaygeorge) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java:7435]: Would this test fail without the above changes. Essentially is this a test that exposes the bug.?

**Review by ajaygeorge — APPROVED** [review]
LGTM % a comment. 
Also let us add a global flag to control the rollout of the new code path if possible.

**Comment by ajaygeorge** [comment]
Please fix the checksytyle violation as well.

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java (hunk 1 above)
Comment (feilong-liu) [reply presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java:7435]: Verified that it will fail without change in this PR

**Review by kaikalur — APPROVED** [review]

**Review by ajaygeorge — APPROVED** [review]

**Pull request merged**